libraries/ms-drivers/inc/adt_7470_fan_controller.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "adc.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "soft_timer.h"
+
+typedef void (*Adt7470DataCallback)();
+
+typedef struct {
+  I2CPort i2c_port;
+  GpioAddress *fan_1_pin;
+  GpioAddress *fan_2_pin;
+  GpioAddress *fan_3_pin;
+  GpioAddress *fan_4_pin;
+  uint32_t interval_us;
+  Adt7470DataCallback callback;  // set to NULL for no callback
+  void *callback_context;
+} Adt7470Settings;
+
+typedef struct {
+  GpioAddress *fan_1_pin;
+  GpioAddress *fan_2_pin;
+  GpioAddress *fan_3_pin;
+  GpioAddress *fan_4_pin;
+  uint32_t interval_us;
+  SoftTimerId timer_id;
+  Adt7470DataCallback callback;
+  void *callback_context;
+} Adt7470Storage;
+
+// probably a better name for these
+typedef enum { FAN_1 = 0, FAN_2, FAN_3, FAN_4 } Fan;
```

Comment:
By convention (and for utility), at the end of enums we add a `NUM_...` value e.g. `NUM_FANS`. This helps iteration so we can do `for (uint8_t i = 0; i < NUM_FANS; i++)` which is a bit more descriptive